examples/next/src/lib/store/use-tx.ts

```diff
@@ -0,0 +1,171 @@
+import { ABIFunction } from '@shazow/whatsabi/lib.types/abi';
+import { create } from 'zustand';
+import { createJSONStorage, persist } from 'zustand/middleware';
+
+import { TxEntry } from '@/lib/types/tx';
+import { TEVM_PREFIX } from '@/lib/local-storage';
+
+import { getFunctionId } from '../utils';
+
+/* ---------------------------------- TYPES --------------------------------- */
+// Input values
+type InputValues = Record<
+  string,
+  { args: Record<number, unknown>; value: string }
+>;
+
+type TxInitialState = {
+  // Tx history
+  txHistory: Record<number, TxEntry[]>; // chainId -> txs
+  // Tx processing
+  processing: string;
+  // Input values
+  inputValues: InputValues;
+
+  isHydrated: boolean;
+};
+
+type TxSetState = {
+  // Tx history
+  saveTx: (chainId: number, tx: Omit<TxEntry, 'id'>) => void;
+  resetTxs: (chainId: number) => void;
+  getTxs: (chainId: number) => TxEntry[];
+  // Tx processing
+  setProcessing: (value: string) => void;
+  // Handling inputs
+  updateInputValue: (id: string, index: number, value: unknown) => void;
+  initializeInputs: (abi: ABIFunction[]) => void; // we eliminated events
+  resetInputs: () => void;
+
+  hydrate: () => void;
+};
+
+type TxStore = TxInitialState & TxSetState;
+
+/* ---------------------------------- STORE --------------------------------- */
+/**
+ * @notice A store to manage the transactions history, and input values for making transactions
+ */
+export const useTxStore = create<TxStore>()(
```

Comment:
We should remove this and instead use the client blockchain to store tx. I will update state persister to persist the tx history instead of only the state